cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_avx2.hpp

```diff
@@ -271,17 +271,20 @@ vertex_similarity_result call_jaccard_default_kernel_avx2(
     auto g_edge_offsets = g->_edge_offsets.data();
     auto g_vertex_neighbors = g->_vertex_neighbors.data();
     auto g_degrees = g->_degrees.data();
-    const std::int32_t row_begin = static_cast<std::int32_t>(desc.get_row_range_begin());
-    const auto row_end = static_cast<std::int32_t>(desc.get_row_range_end());
-    const auto column_begin = static_cast<std::int32_t>(desc.get_column_range_begin());
-    const auto column_end = static_cast<std::int32_t>(desc.get_column_range_end());
-    const auto number_elements_in_block = (row_end - row_begin) * (column_end - column_begin);
-    const size_t max_block_size =
-        compute_max_block_size(row_begin, row_end, column_begin, column_end);
+    const auto row_begin =
+        oneapi::dal::detail::integral_cast<std::int32_t>(desc.get_row_range_begin());
+    const auto row_end = oneapi::dal::detail::integral_cast<std::int32_t>(desc.get_row_range_end());
+    const auto column_begin =
+        oneapi::dal::detail::integral_cast<std::int32_t>(desc.get_column_range_begin());
```

Comment:
`integral_cast` does not throw exception if the cast fails. Is there corresponding check that throws exception for example in the beginning of algorithm or check preconditions?

Reply:
The same question for other `integral_cast`'s

Reply:
Integral cast is based on ONEDAL_ASSERT. That is what we want in Jaccard kernel due to perf overheads of block API.